pkg/pool-weighted/test/BaseManagedPoolFactory.test.ts

```diff
@@ -34,16 +36,23 @@ describe('BaseManagedPoolFactory', function () {
   let createTime: BigNumber;
 
   before('setup signers', async () => {
-    [, manager, assetManager] = await ethers.getSigners();
+    [, admin, manager, assetManager] = await ethers.getSigners();
   });
 
   sharedBeforeEach('deploy factory & tokens', async () => {
-    vault = await Vault.create();
+    vault = await Vault.create({ admin });
 
     factory = await deploy('BaseManagedPoolFactory', { args: [vault.address] });
-    createTime = await currentTimestamp();
 
     tokens = await TokenList.create(['MKR', 'DAI', 'SNX', 'BAT'], { sorted: true });
+
+    createTime = await currentTimestamp();
+
+    protocolFeesCollector = factory.aumProtocolFeesCollector();
+  });
+
+  it('factory has the AumProtocolFeesController', async () => {
+    expect(protocolFeesCollector).to.not.equal(ZERO_ADDRESS);
```

Comment:
This test is not doing what you think it is. We're not `await`ing the getter, so the `feeCollector` is an unresolved promise.

Also, this should be called protocolAUMFeeCollector or something - the feeCollector is a different contract.